app/models/package_manager/go/go_mod.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class PackageManager::Go
+  class GoMod
+    attr_reader :parser
+
+    def initialize(mod_contents)
+      @parser = GoModParser.new(mod_contents)
+    end
+
+    delegate :retracted_version_ranges, :canonical_module_name, :dependencies, to: :parser
+
+    def retracted?(given_version)
+      retracted_version_ranges.any? do |version_or_range|
+        if version_or_range.is_a?(Array)
+          begin_range, end_range = version_or_range
+          SemanticRange.gte?(given_version, begin_range) &&
+            SemanticRange.lte?(given_version, end_range)
+        else
+          SemanticRange.satisfies?(given_version, version_or_range)
+        end
+      end
+    end
+  end
```

Comment:
This entire class is acting on a single `GoModParser` object, calling the methods on that object for everything `retracted?` does as well as passing along delegated values from `GoModParser`. WDYT about moving this method to `GoModParser` and removing `GoMod` entirely?

Reply:
I refactored it all as you suggested, except I put it all into `GoMod` instead of `GoModParser` (but that's a simple renaming).